src/dlaf_fortran.f90

```diff
@@ -0,0 +1,476 @@
+
+            import :: c_ptr, c_int, c_double, c_signed_char
+
+            integer(kind=c_signed_char), value :: uplo_
+            integer(kind=c_int), value :: ia_, ja_, n_
+            type(c_ptr), value :: info_
+            integer(kind=c_int), dimension(*) :: desca_
+            type(c_ptr), value :: a_
+         end subroutine pdpotrf_dlaf
+      end interface
+
+      call pdpotrf_dlaf(iachar(uplo, c_signed_char), n, c_loc(a(1, 1)), ia, ja, desca, c_loc(info))
+
+   end subroutine dlaf_pdpotrf
+
+   subroutine dlaf_pcpotrf(uplo, n, a, ia, ja, desca, info)
+      character, intent(in) :: uplo
+      integer, intent(in) :: n
+      complex(kind=sp), dimension(:, :), target, intent(inout) :: a
+      integer, intent(in) :: ia, ja
+      integer, dimension(9), intent(in) :: desca
+      integer, target, intent(out) :: info
+
+      interface
+         subroutine pcpotrf_dlaf(uplo_, n_, a_, ia_, ja_, desca_, info_) &
+            bind(C, name='dlaf_pcpotrf')
+
+            import :: c_ptr, c_int, c_signed_char
+
+            integer(kind=c_signed_char), value :: uplo_
+            integer(kind=c_int), value :: ia_, ja_, n_
+            type(c_ptr), value :: info_
+            integer(kind=c_int), dimension(*) :: desca_
+            type(c_ptr), value :: a_
+         end subroutine pcpotrf_dlaf
+      end interface
+
+      call pcpotrf_dlaf(iachar(uplo, c_signed_char), n, c_loc(a(1, 1)), ia, ja, desca, c_loc(info))
+
+   end subroutine dlaf_pcpotrf
+
+   subroutine dlaf_pzpotrf(uplo, n, a, ia, ja, desca, info)
+      character, intent(in) :: uplo
+      integer, intent(in) :: n
+      complex(kind=dp), dimension(:, :), target, intent(inout) :: a
+      integer, intent(in) :: ia, ja
+      integer, dimension(9), intent(in) :: desca
+      integer, target, intent(out) :: info
+
+      interface
+         subroutine pzpotrf_dlaf(uplo_, n_, a_, ia_, ja_, desca_, info_) &
+            bind(C, name='dlaf_pzpotrf')
+
+            import :: c_ptr, c_int, c_signed_char
+
+            integer(kind=c_signed_char), value :: uplo_
+            integer(kind=c_int), value :: ia_, ja_, n_
+            type(c_ptr), value :: info_
+            integer(kind=c_int), dimension(*) :: desca_
+            type(c_ptr), value :: a_
+         end subroutine pzpotrf_dlaf
+      end interface
+
+      call pzpotrf_dlaf(iachar(uplo, c_signed_char), n, c_loc(a(1, 1)), ia, ja, desca, c_loc(info))
+
+   end subroutine dlaf_pzpotrf
+
+   subroutine dlaf_pssyevd(uplo, n, a, ia, ja, desca, w, z, iz, jz, descz, info)
+      character, intent(in) :: uplo
+      integer, intent(in) :: n, ia, ja, iz, jz
+      integer, dimension(9), intent(in) :: desca, descz
+      integer, target, intent(out) :: info
+      real(kind=sp), dimension(:, :), target, intent(inout) :: a, z
+      real(kind=sp), dimension(:), target, intent(out) :: w
+
+      interface
+         subroutine pssyevd_dlaf(uplo_, n_, a_, ia_, ja_, desca_, w_, z_, iz_, jz_, descz_, info_) &
```

Comment:
Just out of ignorance, is there a particular reason for the naming scheme with `subroutine dlaf_x` and then in the `interface` `subroutine x_dlaf`?

Reply:
Not really. It's a personal preference for auto-completion (i.e. if I'm integrating DLAF I prefer to autocomplete on `dlaf` to see what's there), but happy to change it if the consensus is to prefer the other way around.

Reply:
No preference for me, I was just curious.

Reply:
Right, my question was indeed about why in one case `dlaf` is a prefix and in the other it's a suffix. It makes sense that it's a prefix as a "namespace", in the same way the C-API is prefixed with `dlaf` in DLA-Future. And then I was simply wondering why not calling the other one something like `dlaf_x_interface` or `dlaf_x_` or `dlaf_x_someothersuffix`. But as long as this latter thing (that is at the moment called `x_dlaf`) is an internal detail I don't mind either way.

Reply:
Yes, it's an internal detail. I have no good justification for `x_dlaf`. A common pattern I've see is to use `dlaf_*_c`, I can convert to that if it's preferred.

Reply:
Yes, I would go either for `dlaf_*_c` or `dlaf_c_*`

Reply:
I changed the naming convention to `dlaf_*_c`.